homeassistant/components/prometheus/__init__.py

```diff
@@ -248,6 +242,24 @@ def handle_state(self, state: State) -> None:
         )
         last_updated_time_seconds.labels(**labels).set(state.last_updated.timestamp())
 
+        if state.state in IGNORED_STATES:
+            ignored_metric_names = set()
+            for metric in (state_change, entity_available, last_updated_time_seconds):
+                for sample in cast(list[prometheus_client.Metric], metric.collect())[
+                    0
+                ].samples:
+                    ignored_metric_names.add(sample.name)
```

Comment:
I don't quite follow why we need to look at samples here, rather than simply pass `(state_change, entity_available, last_updated_time_seconds)` to `_remove_labelsets`?

Reply:
Thank you for pointing out my confusion. The other comment helped a lot too.